packages/tools/devtools/devtools-view/src/components/data-visualization/EditableView.tsx

```diff
@@ -0,0 +1,364 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import React from "react";
+
+import {
+	Button,
+	Dropdown,
+	DropdownProps,
+	Input,
+	InputOnChangeData,
+	Option,
+	makeStyles,
+	shorthands,
+	tokens,
+} from "@fluentui/react-components";
+import {
+	DataEdit,
+	EditData,
+	EditType,
+	FluidObjectValueNode,
+} from "@fluid-experimental/devtools-core";
+
+import { useMessageRelay } from "../../MessageRelayContext";
+import { TreeHeader } from "./TreeHeader";
+
+/**
+ * Input to {@link EditableView}
+ */
+export interface EditableViewProps {
+	node: FluidObjectValueNode;
+	containerKey: string;
+	label: string;
+}
+
+/**
+ * Input to EditableComponents
+ */
+interface EditableComponentProps {
+	node: FluidObjectValueNode;
+	editProp: EditState | undefined;
+	setEditProp: React.Dispatch<React.SetStateAction<EditState | undefined>>;
+	submitChange: (data: EditData) => void;
+}
+
+interface EditState {
+	type: string;
+	value?: EditData;
```

Comment:
Nit: Since `undefined` means something special / explicit here, it might be nice to make the parameter required.
```suggestion
	value: EditData | undefined;
```

Reply:
This is a bit tricky since value being undefined could mean that the value is undefined, or the edit doesn't have a value yet. What are your thoughts, maybe I missed the point of the original comment. I am not sure how to fix this.

Reply:
Since we always immediately submit an edit that makes the value `null` or `undefined` (upon clicking the option in the dropdown), we can probably type `edit.value` as just `Serializable<unknown> | undefined`, and use `undefined` to mean "not set yet".